nifi-framework-api/src/main/java/org/apache/nifi/action/FlowAction.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.action;
+
+import java.util.Map;
+
+/**
+ * An interface that represents an action that can be taken on a flow
+ */
+public interface FlowAction {
+    Map<String, String> getAttributes();
+
+    void putAllAttributes(Map<String, String> properties);
+
+    String getAttribute(String key);
+
+    void putAttribute(String key, String value);
```

Comment:
These methods should have documentation. As a `FlowAction` is something to be consumed, I recommend simplifying the interface to just `getAttributes()`, the convenience `getAttribute()` method is not necessary.